lib/std/mem.zig

```diff
@@ -2190,6 +2377,13 @@ test "alignForward" {
     testing.expect(alignForward(17, 8) == 24);
 }
 
+pub fn alignBackwardAnyAlign(i: usize, alignment: usize) usize {
```

Comment:
Can you add doc comments to explain why this is needed rather than just using `alignBackward`?

Looks like it removes the restriction " /// The alignment must be a power of 2 and greater than 0. " ?

Is this necessary at the callsites? Or would it be appropriate to add that restriction to the callsites as well, and have them call `alignBackward` directly?

Reply:
This is necessary because arrays are aligning by element size which don't have to be powers of 2.  When I was using `alignBackwards` originally, there were cases where size of structs were not a power of 2 (i.e. I saw `56`).